lib/Doctrine/Migrations/Tools/Console/Command/VersionCommand.php

```diff
@@ -168,10 +168,29 @@ private function markVersions(InputInterface $input, OutputInterface $output) :
      * @throws VersionDoesNotExist
      * @throws UnknownMigrationVersion
      */
-    private function mark(OutputInterface $output, string $version, bool $all = false) : void
+    private function mark(InputInterface $input, OutputInterface $output, string $version, bool $all = false) : void
     {
         if (! $this->migrationRepository->hasVersion($version)) {
-            throw UnknownMigrationVersion::new($version);
+            if ((bool) $input->getOption('delete') === false) {
```

Comment:
I'm not sure if VersionCommand already has a test or not but we need to test this functionality/logic here too.